### PR TITLE
fix empty dps API request edge case

### PR DESCRIPTION
**Pull request by haakonvt**
## Description
make pytest show output from tests on failure + fix several flawed datapoints test.

For SDK users:
### Fixed
- An (extreme) edge case where an empty, unnecessary API request for datapoints would be sent leading to a `CogniteAPIError`.
- Certain granularity inputs (when using the `DatapointsAPI`) no longer causes a `ValueError` to be raised with confusing/wrong wording.

## Checklist:
- [x] Tests added/updated.

### CODE REVIEW

**Comment by haakonvt** [comment]
Two error modes here:
1. hourly aggregate being far too wide to return up to limit
2. `interpolation` really hating on the fixed bursty nature of one of the two test time series, see screenshot below where it returns exactly 0 datapoints:

----
image

**Comment by haakonvt** [comment]
pytest changes:
``` 
-r chars              Show extra test summary info as specified by chars:
                        (f)ailed, (E)rror, (s)kipped, (x)failed, (X)passed,
                        (p)assed, (P)assed with output, (a)ll except passed
                        (p/P), or (A)ll. (w)arnings are enabled by default
                        (see --disable-warnings), 'N' can be used to reset
                        the list. (default: 'fE').
```

**Review by erlendvollset — APPROVED** [review]
LGTM!

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,11 @@ Changes are grouped as follows
 - `Fixed` for any bug fixes.
 - `Security` in case of vulnerabilities.
 
+## [7.17.1] - 2024-02-02
+### Fixed
+- An (extreme) edge case where an empty, unneccesary API request for datapoints would be sent leading to a `CogniteAPIError`.
+- Certain granularity inputs (when using the `DatapointsAPI`) no longer causes a `ValueError` to be raised with confusing/wrong wording.
```
Comment (haakonvt) [inline CHANGELOG.md:23]:
```suggestion
- An (extreme) edge case where an empty, unnecessary API request for datapoints would be sent leading to a `CogniteAPIError`.
- Certain granularity inputs (when using the `DatapointsAPI`) no longer cause a `ValueError` to be raised with confusing/wrong wording.
```

File: cognite/client/utils/_time.py [hunk 2]
```diff
@@ -120,11 +124,6 @@ def granularity_to_ms(granularity: str) -> int:
     return ms
 
 
-def granularity_unit_to_ms(granularity: str) -> int:
-    granularity = re.sub(r"^\d+", "1", granularity)
-    return granularity_to_ms(granularity)
```
Comment (haakonvt) [inline cognite/client/utils/_time.py:125]: With input of something like `"30x"` when `x` is an invalid unit, or `"23"` (missing time unit), the error message would say that the input was `"1x"` or just `"1"`
